presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -113,7 +113,30 @@ General Aggregate Functions
     for each input value. In addition to taking the input value, ``inputFunction``
     takes the current state, initially ``initialState``, and returns the new state.
     ``combineFunction`` will be invoked to combine two states into a new state.
-    The final state is returned::
+    The final state is returned. Throws an error if ``initialState`` is NULL.
+    The behavior is undefined if ``inputFunction`` or ``combineFunction`` return a NULL.
+
+    Take care when designing ``initialState``, ``inputFunction`` and ``combineFunction``.
+    These need to support evaluating aggregation in a distributed manner using partial
+    aggregation on many nodes, followed by shuffle over group-by keys, followed by
+    final aggregation. Given a set of all possible values of state, make sure that
+    combineFunction is `commutative <https://en.wikipedia.org/wiki/Commutative_property>`_
+    and `associative <https://en.wikipedia.org/wiki/Associative_property>`_
+    operation with initialState as the
+    `identity <https://en.wikipedia.org/wiki/Identity_element>`_ value.
+
+     combineFunction(s, initialState) = s for any s
+
+     combineFunction(s1, s2) = combineFunction(s2, s1) for any s1 and s2
+
+     combineFunction(s1, combineFunction(s2, s3)) = combineFunction(combineFunction(s1, s2), s3) for any s1, s2, s3
+
+    In addition, make sure that the following holds for the inputFunction:
+
+     inputFunction(inputFunction(initialState, x), y) = combineFunction(inputFunction(initialState, x), inputFunction(initialState, y)) for any x and y
+
+    Check out `Velox blog post about reduce_agg <https://velox-lib.io/blog/reduce-agg>`_ for
```

Comment:
Let's just directly add whatever context is necessary directly in the technical documentation.  Although it might seem hard to believe, you're updating documentation that hasn't been touched since *2018*, so we're looking at a change rate of 6 years.  It definitely seems plausible that Velox site structure may change in the span of 6 years and generate broken links.

Reply:
This is a very complex topic, so I don't think I can summarize it "quickly". I'm happy to remove the link if you think it is better without.

Reply:
I would remove, as I think what's written here is sufficient for users.  The blog post is very nice but not very relevant for user-facing technical documentation.

Reply:
@tdcmeehan Removed. Would you take another look?